src/journal-remote/journal-upload.c

```diff
@@ -423,11 +551,40 @@ static int setup_uploader(Uploader *u, const char *url, const char *state_file)
         if (r < 0)
                 return log_error_errno(r, "Failed to install SIGINT/SIGTERM handlers: %m");
 
+        struct sigevent sev;
+        sev.sigev_notify = SIGEV_THREAD;
+        sev.sigev_value.sival_ptr = u;
+        sev.sigev_notify_function = &finish_uploading;
+        sev.sigev_notify_attributes = 0;
```

Comment:
sorry, what is this? nope. we do not use threads for this, and unix signals are the worst. i don#t get this. this has to go.